specs/executionAPISpec.md

```diff
@@ -0,0 +1,530 @@
+[3 Errors](#3-errors)\
+[Appendix](#appendix)
+# 1 Introduction 
+## 1.1 Overview
+The Ethereum execution layer API is one of the key components of Ethereum. It acts as an intermediary between the users of Ethereum and the execution layer where the blocks are created and executed. It provides a way for users to send transactions to the blockchain, query data from the current and historical states of the blockchain, and monitor events and changes happening within the blockchain
+
+The purpose of this document is to act as a centralized source of information regarding the functional and non-functional requirements for Ethereum's execution layer API. This document is intended for development teams that are planning on implementing a version of the execution layer API. This document would also be beneficial to but is not intended for anyone interested in learning how the user interacts with Ethereum clients and at the most basic level. 
+## 1.2 Terminology
+|Term| Definition|
+|---|---|
+|Account||
+|Address||
+|Block||
+|Blockchain||
+|call||
+|client| Software that allows users to run an instance of the Ethereum blockchain|
+|cryptography||
+|Ether|Cryptographic token created as a reward for participating in securing and running the Ethereum blockchain. Ether is also used to pay for all transactions on the Ethereum blockchain.|
+|EVM||
+|Execution layer||
+|gas||
+|genisis block||
+|hash||
+|header||
+|merkle tree||
+|mining||
+|network||
+|node| A running client connected to the Ethereum blockchain.|
+|nonce| An integer corresponding to the number of transactions an account has made on the blockchain.|
+|peer nodes||
+|private key||
+|public key||
+|rewards||
+|signing||
+|Smart Contract| Code|
+|state||
+|storage slot||
+|token||
+|transaction||
+|Uncle block| A block that was |
+|user||
+|Wei|The smallest unit of Ether where **1Eth = 1e18Wei**|
+## 1.3 References
+### 1.3.1 Normative References
+### 1.3.2 Non-normative References
+## 1.4 Document Conventions
+### 1.4.1 Input Parameters
+Input parameters will be displayed like `this` when referred to in the document.
+### 1.4.2 Not Specified vs Null vs Empty
+* Not specified describes when a parameter is not part of the call.
+* Null describes when a parameter is part of the call, but has the value of null.
+* Empty describes when a parameter is part of the call, but has only an empty string.
+
+An example where `gas` is not specified `value` is empty, and `data` is null.
+```
+{
+	"jsonrpc": "2.0",
+	"method": "eth_sendTransaction",
+	"params": [
+		{
+			"from": "0x3a5509015e0193adf435a761a6ce160f900034b5",
+			"to": "0xe64fac7f3df5ab44333ad3d3eb3fb68be43f2e8c",
+			"value": "",
+			"data": null
+		}
+	],
+	"id": 1
+}
+```
+### 1.4.3 Default Block Parameter
+Some endpoints use the `defaultBlockParameter` to specify the block that the data is being requested from.
+The `defaultBlockParameter` allows the following options to specify a block:
```

Comment:
```suggestion
Some endpoints use the `blockParameter` to specify the block that the data is being requested from.
The `blockParameter` allows the following options to specify a block:
```
This variable should be renamed throughout the document.  `default` prefix here doesn't make sense, as it is a type and types don't have defaults.